### DISCUSSION TITLE
Cannot override `css_class_name` on class instantiation

### DISCUSSION THREAD

**Discussion by adrien-delhorme** [codingjoe/django-select2#221] [Q&A]
Hi,

https://github.com/codingjoe/django-select2/blob/ed1af309908d1c8c94a189209b0a40479ee95320/django_select2/forms.py#L83

The `css_class_name` attribute on all widgets classes inheriting `Select2Mixin` cannot be overridden when instantiating the class. Because there is no 
```py
def __init__(self, *args, **kwargs):
    ...
    this.css_class_name = kwargs.get("css_class_name")
    ...
```
in any `__init__()`.

The only way to override `css_class_name` is to create a new class inheriting the desired widget: not very handy.

Is this intentional? Can we improve this?

**Comment by codingjoe** [comment]
Hi @adrien-delhorme,

Yes, that was intentional. The thought behind it was, to have a separate Python class for a different HTML element class. Is there any particular reason, why doesn't inheritance work well in your use case? I am curious. There are so many ways to do things.

Cheers!
Joe

**Reply by adrien-delhorme** [reply]
Hello,

I just needed to override the `css_class_name` attribute of the widget and I thought from the documentation, I could just do:
```py
class MyForm(forms.ModelForm):
    my_field = forms.MultipleChoiceField(
        label="My label",
        widget=ModelSelect2MultipleWidget(css_class_name="my-class-name"),
    )
```

but actually, I had to do:
```py
class MyCustomClassModelSelect2MultipleWidget(ModelSelect2MultipleWidget):
    css_class_name = "my-class-name"

class MyForm(forms.ModelForm):
    my_field = forms.MultipleChoiceField(
        label="My label",
       widget=MyCustomClassModelSelect2MultipleWidget(),
    )
```

My expectation of how it was supposed to work was wrong and it took me several minutes 😅 to understand why it was not working. Maybe I am the only one to think this way.
